src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h

```diff
@@ -43,13 +43,57 @@
 #include <thread>
 #include <tuple>
 #include <sys/wait.h>
+#include<filesystem>
 #ifndef _WINDOWS
 #include <dlfcn.h>
 #endif
 
 namespace xclcpuemhal2 {
   using key_type = xrt_core::query::key_type;
   const uint64_t MEMSIZE = 0x0000000080000000;
+  struct sParseLog
+  {
+    std::ifstream file;
+    std::string mFileName;
+    std::atomic_bool mFileExists;
+
+    sParseLog(const std::string& iDeviceLog):mFileName(iDeviceLog) {
+      mFileExists.store(false);    
+    }  
+
+    ~sParseLog() = default;
+    //**********************************************************************************//
+    /**
+    * SearchString(std::string&) - Searches for a string 
+    *
+    * @matchString:    string to match
+    * 
+    */
+    void SearchString(std::string &matchString) {
+      std::string line;
+      while (std::getline(file, line)){
+        if (line.find(matchString) != std::string::npos)
+          std::cout << "Received request to end the application. Press Cntrl+C to exit the application." << std::endl; 
+      }
+    }
+
+    void parseLog() {
+      if (not mFileExists.load()) {
+        if (std::filesystem::exists(mFileName)) {
+          file.open(mFileName);
+          if (file.is_open())
+            mFileExists.store(true);
+        }
+      }
+
+      if(mFileExists.load()) {
+        std::string matchString = "received request to end simulation from connected initiator";
```

Comment:
Should this protocol string declared as a `constexpr` somewhere more globally as a description of some protocol?

Reply:
Renamed matchString to endOfSimulationString and declared it under xclcpuemhal2 namespace.